src/daemon/daemon_rpc.cpp

```diff
@@ -84,6 +84,18 @@ auto make_server(const std::string& server_address, mp::RpcConnectionType conn_t
 
     return server;
 }
+
+template <typename OperationSignal>
+grpc::Status emit_signal_and_wait_for_result(OperationSignal operation_signal)
+{
+    std::promise<grpc::Status> status_promise;
+    auto status_future = status_promise.get_future();
+    emit operation_signal(&status_promise);
+
+    status_future.wait();
```

Comment:
This is a bit redundant since `get` already waits.

Reply:
I'm not sure what you mean here.  Do you mean `get_future()` is supposed to wait (it doesn't)? Or do you mean there is a better method to use than `wait()`?

Reply:
I was referring to the following get (line 98), which itself waits:

> [...] effectively calls wait() in order to wait for the result

(in https://en.cppreference.com/w/cpp/thread/future/get)

Reply:
Ah, ok, lol, I didn't see the `get()` call within the comments and forgot that was there.  I will fix:)

Reply:
Yeah, github's review interface mixes everything up

Reply:
Done